File: tourism/src/main/java/com/tourism/Manager.java
package com.tourism;

/**
 * Клас для представлення менеджера турфірми
 */
public class Manager extends TravelSubject {
    public Manager(String name) {
        super(name);
    }

    @Override
    public void showRole() {
        System.out.println(name + " — Менеджер турфірми.");
    }

    @Override
    public void makeReservation(String roomType) {
        System.out.println("Менеджер " + name + " підбирає пакет з номером: " + roomType);
    }

    @Override
    public void cancelReservation() {
        System.out.println("Менеджер " + name + " скасовує бронювання клієнта.");
    }

    public void approvePackage() {
        System.out.println("Менеджер погоджує умови пакету.");
    }
}

File: tourism-python/src/tourism/manager.py
from tourism.subjects import TravelSubject


class Manager(TravelSubject):
    """
    Клас для представлення менеджера турфірми
    """

    def show_role(self):
        print(f"{self.name} — Менеджер турфірми.")

    def make_reservation(self, room_type: str):
        print(f"Менеджер {self.name} підбирає пакет з номером: {room_type}")

    def cancel_reservation(self):
        print(f"Менеджер {self.name} скасовує бронювання клієнта.")

    def approve_package(self):
        print("Менеджер погоджує умови пакету.")
